### PR TITLE
docs: add a guide on clearing migrations (close #5372)

### LINKED ISSUE #5372: docs: add info on clearing migrations
There's info on how to clear / reset metadata, but not migrations.

The relevant information has been shared via a blog post, and should be added to the docs:
https://hasura.io/blog/resetting-hasura-migrations/

```
TRUNCATE hdb_catalog.schema_migrations;
```

### PR DESCRIPTION
**Pull request by codeliezel**
### Description
This PR adds information on clearing migrations to the docs
### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components
- [x] Docs

### Related Issues
https://github.com/hasura/graphql-engine/issues/5372

### Old PR
https://github.com/hasura/graphql-engine/pull/6015

### Affected pages
https://deploy-preview-6224--hasura-docs.netlify.app/graphql/core/migrations/advanced/resetting-migrations.html

### CODE REVIEW

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+------------
+
+This page explains how to reset the state of migrations and create new migrations from the state that is on the server. This can be useful if the current migration state on your local machine is corrupted.
+
+Step 1: Delete the local migrations
+-----------------------------------
+
+From your local project directory, delete all the contents in the ``migrations`` directory using the following command:
+
+.. code-block:: bash
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:25]: Command doesn't show

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   rm -r migrations/*
+Step 2: Reset the migration history on the server
+-------------------------------------------------
+
+On the SQL tab of the Hasura console, run the following command:
+
+.. code-block:: bash
+   TRUNCATE hdb_catalog.schema_migrations;
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:33]: Command doesn't show

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+.. code-block:: bash
+   ## create migration files - "this will only export public schema from postgres"
+   hasura migrate create "init" --from-server
+.. code-block:: bash
+   ## note down the version
+   ## mark the migration as applied on this server
+   hasura migrate apply --version "<version>" --skip-execution
+This conversation was marked as resolved by codeliezel
+.. note::
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:47]: Note doesn't show correctly

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+   If you are using schemas other than ``public``, use the ``--schema <schema_name>`` flag to indicate each one of them in the create command. This flag can be used multiple times.
+
+
+Step 4: Verify the status of the migrations
+-------------------------------------------
+
+Run the following command to verify the migration status:
+
+.. code-block:: bash
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:57]: Command doesn't who

**Review by marionschleifer — CHANGES_REQUESTED** [review]

**Review by marionschleifer — APPROVED** [review]

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+From your local project directory, delete all the contents in the ``migrations`` directory using the following command:
+
+.. code-block:: bash
+
+   rm -r migrations/*
+
+.. note::
+
+   Before deleting the local migrations, you might consider storing them in a backup folder in case you need them later using the following command:
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:32]: This note should be in the beginning. If someone follows the tutorial, they might delete the `migrations` folder and only then see that they should take a backup.

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+.. code-block:: bash
+   
+   mv -r migrations migrations_bckup
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:36]: It looks a little weird to have this **after** a note. We can either remove the note or put the command in the note also.

**Review by marionschleifer — CHANGES_REQUESTED** [review]

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst (hunk 5 above)
Comment (rikinsk) [reply docs/graphql/core/migrations/advanced/resetting-migrations.rst:32]: I would actually remove the `rm -r` command completely and just suggest the `mv -r` command instead. The title of the section can remain "delete local migrations" but the operation we should do is to move the "migration" dir to "migration_bckup".

Comment (codeliezel) [reply docs/graphql/core/migrations/advanced/resetting-migrations.rst:32]: thanks! makes sense @rikinsk

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst (hunk 6 above)
Comment (codeliezel) [reply docs/graphql/core/migrations/advanced/resetting-migrations.rst:36]: Alright @marionschleifer

File: docs/graphql/core/migrations/advanced/resetting-migrations.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+This can be useful if the current migration state on your local machine is corrupted.
+
+Step 1: Delete the local migrations
+-----------------------------------
+
+.. note::
+
+   Before deleting the local migrations, you might consider storing them in a backup folder in case you need them later using this command: ``mv -r migrations migrations_bckup``
+   
+Also, if you do not wish to create a back up folder, you can use the following command to delete all the contents in the ``migrations`` directory.
```
Comment (marionschleifer) [inline docs/graphql/core/migrations/advanced/resetting-migrations.rst:28]: This is not either or. We have to delete the contents anyway to clear migrations. Regardless of if we create a backup folder.

**Review by marionschleifer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**